pkg/apply/apply.go

```diff
@@ -86,9 +102,53 @@ func ApplyObject(ctx context.Context, client *cnoclient.Client, obj Object, subc
 		log.Printf("could not encode %s for apply", objDesc)
 		return fmt.Errorf("could not encode for patching: %w", err)
 	}
-	if _, err := client.Dynamic().Resource(rm.Resource).Namespace(namespace).Patch(ctx, name, types.ApplyPatchType, data, patchOptions); err != nil {
+	if _, err := clusterClient.Dynamic().Resource(rm.Resource).Namespace(namespace).Patch(ctx, name, types.ApplyPatchType, data, patchOptions); err != nil {
 		return fmt.Errorf("failed to apply / update %s: %w", objDesc, err)
 	}
 	log.Printf("Apply / Create of %s was successful", objDesc)
 	return nil
 }
+
+func getObjectCopy(ctx context.Context, obj Object, client *cnoclient.Client) (Object, error) {
+	anno := obj.GetAnnotations()
+	if anno == nil {
+		return nil, fmt.Errorf("no annotations defined")
+	}
+
+	parts := strings.Split(anno[names.CopyFromAnnotation], "/")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("'%s' annotation is invalid, expected: ClusterName/Namespace/ResourceName, got: %s", names.CopyFromAnnotation, anno[names.CopyFromAnnotation])
+	}
+
+	clusterName, namespace, name := parts[0], parts[1], parts[2]
+
+	cli := client.ClientFor(clusterName)
+	if cli == nil {
+		return nil, fmt.Errorf("failed to find the client for %s", clusterName)
+	}
+
+	ret := &unstructured.Unstructured{}
+	ret.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+
+	err := cli.CRClient().Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, ret)
+	if err != nil {
+		return nil, err
+	}
+
+	ret.SetNamespace(obj.GetNamespace())
+	ret.SetName(obj.GetName())
+	ret.SetClusterName(obj.GetClusterName())
+	ret.SetAnnotations(obj.GetAnnotations())
+	ret.SetLabels(obj.GetLabels())
+	ret.SetOwnerReferences(obj.GetOwnerReferences())
+
+	ret.SetCreationTimestamp(obj.GetCreationTimestamp())
+	ret.SetSelfLink(obj.GetSelfLink())
+	ret.SetGeneration(obj.GetGeneration())
+	ret.SetUID(obj.GetUID())
+	ret.SetResourceVersion(obj.GetResourceVersion())
+	ret.SetManagedFields(obj.GetManagedFields())
```

Comment:
We cant touch managed fields.

Reply:
Aha, this is subtle. Fun. We need to be careful about all of the ObjectMeta fields.
- namespace, name: from the requesting object
- annotations: merge from requesting and source objects?
- labels: from the requesting object?
- owners: from the requesting object
- finalizers: from the requesting object (or just don't support setting them)

all other ObjectMeta fields should be empty.

Reply:
Oh, ResourceVersion should be from the requesting object too.